sematic/ee/plugins/external_resource/ray/checkpoint.py

```diff
@@ -0,0 +1,44 @@
+# Sematic
+from sematic.types.registry import register_to_json_encodable_summary
+
+try:
+    # Third-party
+    import torch
+    from ray.train.torch import TorchCheckpoint  # type: ignore
+except Exception:
+    # Why "Exception" instead of just "ImportError"? Because
+    # torch can raise non-standard errors when it fails to import
+
+    # type is not available, so it just won't be registered.
+    # Note that the type is not available in "bare ray", but
+    # rather only in ray AIR.
```

Comment:
`requirements.in` specifies the requirements bazel will use for us while we develop Sematic, it does not dictate (a) what versions of the dependencies our wheel requires (which is determined [here](https://github.com/sematic-ai/sematic/blob/main/sematic/ee/extras.bzl#L10)) nor (b) what versions of the dependencies our users have installed. Our wheel does not require the `air` capabilities of `ray`, so users may or may not have them installed. So this is saying: "If the *user* has Ray AIR installed, we will register this handler, otherwise we won't "